lib/system.nim

```diff
@@ -2320,6 +2320,15 @@ when notJSnotNims:
   proc setControlCHook*(hook: proc () {.noconv.})
     ## Allows you to override the behaviour of your application when CTRL+C
     ## is pressed. Only one such hook is supported.
+    ## Example:
+    ##
+    ## .. code-block:: Nim
+    ##   proc ctrlc() {.noconv.} =
+    ##     echo "Ctrl+C fired!"
+    ##     # do clean up stuff
+    ##     quit()
+    ##
+    ##   setControlCHook(ctrlc)
```

Comment:
For the future: `runnableExamples` are preferred for examples, since they're tested when the documentation is generated.

Reply:
Thank you - I was advised that in this case it would be tricky to send a ctrl-c for testing?

Reply:
You can use `runnableExamples("-r:off")`. But `runnableExamples` seems not to be used for forward declare.

Reply:
You can use runnableExamples("-r:off"). But runnableExamples seems not to be used for forward declare.

Reply:
Thank you once again xflywind. I don't know what "But runnableExamples seems not to be used for forward declare" means?

Reply:
It simply doesn't work for forward declare.


```nim
  proc setControlCHook*(hook: proc () {.noconv.})
    ## Allows you to override the behaviour of your application when CTRL+C
    ## is pressed. Only one such hook is supported.
    runnableExamples("-r:off"):
      proc ctrlc() {.noconv.} =
        echo "Ctrl+C fired!"
        # do clean up stuff
        quit()
      setControlCHook(ctrlc)
```
=> Error: invalid indentation